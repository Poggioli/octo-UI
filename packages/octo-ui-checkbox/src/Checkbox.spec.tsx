import { render } from "@testing-library/react";
import {
  CheckboxGroup,
  CheckboxItem,
  CheckboxItemContainer,
  CheckboxItemLabel
} from "./Checkbox";

describe("<Checkbox />", () => {
  it(`GIVEN a <Checkbox />
      WHEN render is called
      THEN should render a <Checkbox />`, () => {
    const rendered = render(
      <CheckboxGroup>
        <CheckboxItemContainer>
          <CheckboxItem id="fish" name="animal" value="fish" />
          <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
        </CheckboxItemContainer>
      </CheckboxGroup>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Checkbox />
      WHEN it's not inside a <CheckboxGroup />
      THEN throw an error`, () => {
    expect(() =>
      render(
        <CheckboxItemContainer>
          <CheckboxItem id="fish" name="animal" value="fish" />
          <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
        </CheckboxItemContainer>
      )
    ).toThrowError("`CheckboxItem` must be used within `CheckboxGroup`");
  });

  it(`GIVEN a <CheckboxGroup />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CheckboxGroup.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <CheckboxItemContainer />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CheckboxItemContainer.toString()).toBe(".octo-UI-c-hrAFsB");
  });

  it(`GIVEN a <CheckboxItem />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CheckboxItem.toString()).toBe(".octo-UI-c-fwkXIE");
  });

  it(`GIVEN a <CheckboxItemLabel />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CheckboxItemLabel.toString()).toBe(".octo-UI-c-gHWBzz");
  });

  describe("GIVEN a <CheckboxGroup/> whit disabled props", () => {
    it(`WHEN disabled is true
        THEN should disable all checkboxes`, () => {
      const rendered = render(
        <CheckboxGroup disabled>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeDisabled();
      expect(dogCheckbox).toBeDisabled();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN disabled is false
        AND fish checkbox is disabled
        THEN should render dog as enabled
        AND fish as disabled`, () => {
      const rendered = render(
        <CheckboxGroup disabled={false}>
          <CheckboxItemContainer>
            <CheckboxItem disabled id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeDisabled();
      expect(dogCheckbox).toBeEnabled();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN disabled is not setted
        THEN should enable all enabled`, () => {
      const rendered = render(
        <CheckboxGroup>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeEnabled();
      expect(dogCheckbox).toBeEnabled();
      expect(rendered.container).toMatchSnapshot();
    });
  });

  describe("GIVEN a <CheckboxGroup/> whit defaultValue props", () => {
    it(`WHEN defaultValue is undefined
        THEN all checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup defaultValue={undefined}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).not.toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN defaultValue has fish value
        THEN fish checkbox should be checked
        AND dog checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup defaultValue={["fish"]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN defaultValue has all values
        THEN all checkboxes should be checked`, () => {
      const rendered = render(
        <CheckboxGroup defaultValue={["fish", "dog"]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeChecked();
      expect(dogCheckbox).toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN defaultValue is empty
        THEN all checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup defaultValue={[]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).not.toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });
  });

  describe("GIVEN a <CheckboxGroup/> whit value props", () => {
    it(`WHEN value is undefined
        THEN all checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup value={undefined}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).not.toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN value has fish value
        THEN fish checkbox should be checked
        AND dog checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup value={["fish"]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN value has all values
        THEN all checkboxes should be checked`, () => {
      const rendered = render(
        <CheckboxGroup value={["fish", "dog"]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeChecked();
      expect(dogCheckbox).toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN value is empty
        THEN all checkbox should be unchecked`, () => {
      const rendered = render(
        <CheckboxGroup value={[]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).not.toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });

    it(`WHEN defaultValue has dog checked
        AND value doesn't have dog checked 
        THEN dog checkbox should not be checked`, () => {
      const rendered = render(
        <CheckboxGroup defaultValue={["dog", "fish"]} value={["fish"]}>
          <CheckboxItemContainer>
            <CheckboxItem id="fish" name="fish" value="fish" />
            <CheckboxItemLabel htmlFor="fish">Fish</CheckboxItemLabel>
          </CheckboxItemContainer>
          <CheckboxItemContainer>
            <CheckboxItem id="dog" name="dog" value="dog" />
            <CheckboxItemLabel htmlFor="dog">Dog</CheckboxItemLabel>
          </CheckboxItemContainer>
        </CheckboxGroup>
      );
      const fishCheckbox = rendered.getByLabelText("Fish");
      const dogCheckbox = rendered.getByLabelText("Dog");

      expect(fishCheckbox).toBeChecked();
      expect(dogCheckbox).not.toBeChecked();
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
