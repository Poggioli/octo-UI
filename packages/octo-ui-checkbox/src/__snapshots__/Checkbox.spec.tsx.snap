// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> GIVEN a <Checkbox />
      WHEN render is called
      THEN should render a <Checkbox /> 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit defaultValue props WHEN defaultValue has all values
        THEN all checkboxes should be checked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit defaultValue props WHEN defaultValue has fish value
        THEN fish checkbox should be checked
        AND dog checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit defaultValue props WHEN defaultValue is empty
        THEN all checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit defaultValue props WHEN defaultValue is undefined
        THEN all checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit disabled props WHEN disabled is false
        AND fish checkbox is disabled
        THEN should render dog as enabled
        AND fish as disabled 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-disabled=""
        data-state="unchecked"
        disabled=""
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit disabled props WHEN disabled is not setted
        THEN should enable all enabled 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit disabled props WHEN disabled is true
        THEN should disable all checkboxes 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-disabled=""
        data-state="unchecked"
        disabled=""
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-disabled=""
        data-state="unchecked"
        disabled=""
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit value props WHEN defaultValue has dog checked
        AND value doesn't have dog checked 
        THEN dog checkbox should not be checked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit value props WHEN value has all values
        THEN all checkboxes should be checked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit value props WHEN value has fish value
        THEN fish checkbox should be checked
        AND dog checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="true"
        class="octo-UI-c-fwkXIE"
        data-state="checked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      >
        <span
          class="octo-UI-c-eIguWi"
          data-state="checked"
          style="pointer-events: none;"
        >
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.4669 3.72684C11.7558 3.91574 11.8369 4.30308 11.648 4.59198L7.39799 11.092C7.29783 11.2452 7.13556 11.3467 6.95402 11.3699C6.77247 11.3931 6.58989 11.3355 6.45446 11.2124L3.70446 8.71241C3.44905 8.48022 3.43023 8.08494 3.66242 7.82953C3.89461 7.57412 4.28989 7.55529 4.5453 7.78749L6.75292 9.79441L10.6018 3.90792C10.7907 3.61902 11.178 3.53795 11.4669 3.72684Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit value props WHEN value is empty
        THEN all checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;

exports[`<Checkbox /> GIVEN a <CheckboxGroup/> whit value props WHEN value is undefined
        THEN all checkbox should be unchecked 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV octo-UI-c-PJLV-iNpQip-color-unstyled"
  >
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="fish"
        role="checkbox"
        type="button"
        value="fish"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="fish"
      >
        Fish
      </label>
    </div>
    <div
      class="octo-UI-c-hrAFsB"
    >
      <button
        aria-checked="false"
        class="octo-UI-c-fwkXIE"
        data-state="unchecked"
        id="dog"
        role="checkbox"
        type="button"
        value="dog"
      />
      <label
        class="octo-UI-c-gHWBzz octo-UI-c-kwAZbD"
        for="dog"
      >
        Dog
      </label>
    </div>
  </div>
</div>
`;
